library/src/main/java/com/chuckerteam/chucker/internal/support/RequestProcessor.kt

```diff
@@ -0,0 +1,66 @@
+package com.chuckerteam.chucker.internal.support
+
+import android.content.Context
+import com.chuckerteam.chucker.R
+import com.chuckerteam.chucker.api.ChuckerCollector
+import com.chuckerteam.chucker.internal.data.entity.HttpTransaction
+import okhttp3.Request
+import okio.Buffer
+import okio.IOException
+import kotlin.text.Charsets.UTF_8
+
+internal class RequestProcessor(
+    private val context: Context,
+    private val collector: ChuckerCollector,
+    private val maxContentLength: Long,
+) {
+    fun process(request: Request, transaction: HttpTransaction): Request {
+        processMetadata(request, transaction)
+        processBody(request, transaction)
+        collector.onRequestSent(transaction)
+        return request
+    }
+
+    private fun processMetadata(request: Request, transaction: HttpTransaction) {
+        transaction.apply {
+            setRequestHeaders(request.headers)
+            populateUrl(request.url)
+
+            requestDate = System.currentTimeMillis()
+            method = request.method
+            requestContentType = request.body?.contentType()?.toString()
+            requestPayloadSize = request.body?.contentLength()
+        }
+    }
+
+    private fun processBody(request: Request, transaction: HttpTransaction) {
+        val body = request.body ?: return
+
+        val isEncodingSupported = request.headers.hasSupportedContentEncoding
+        if (!isEncodingSupported) {
+            return
+        }
+
+        val limitingSource = try {
+            Buffer().apply { body.writeTo(this) }
+        } catch (e: IOException) {
+            Logger.error("Failed to read request payload", e)
+            return
+        }.uncompress(request.headers).let { LimitingSource(it, maxContentLength) }
+
+        val contentBuffer = Buffer().apply { limitingSource.use { writeAll(it) } }
+        if (!contentBuffer.isProbablyPlainText) {
+            return
+        }
+
+        transaction.isRequestBodyPlainText = true
```

Comment:
_Commenting here but not exactly related to this line:_

Have you verified that this works fine with images (even with truncated ones)? I also believe that `isRequestBodyPlainText` is probably a confusing name within the image scenario. Maybe we can come up with a better name?

Reply:
Not sure I got that. We never processed images in requests so they won't pass through supported encoding check.

Reply:
🤦 I thought we were looking at responses my bad. Closing here.